File: apps/api/app/schemas/discovery.py
from pydantic import BaseModel, Field


class DiscoverySearchRequest(BaseModel):
    query: str = Field(min_length=1, max_length=255)
    location: str = Field(min_length=1, max_length=255)


class DiscoveryCandidate(BaseModel):
    name: str
    website: str | None = None
    phone: str | None = None
    address: str | None = None
    external_id: str
    already_imported: bool = False


class DiscoverySearchResponse(BaseModel):
    candidates: list[DiscoveryCandidate]


class DiscoveryImportRequest(BaseModel):
    candidates: list[DiscoveryCandidate]


class DiscoveryImportResponse(BaseModel):
    created: int
    skipped_duplicate: int

File: apps/web/lib/discovery.ts
import { apiFetch } from "./api-client";

export interface DiscoveryCandidate {
  name: string;
  website: string | null;
  phone: string | null;
  address: string | null;
  external_id: string;
  already_imported: boolean;
}

export interface DiscoverySearchResponse {
  candidates: DiscoveryCandidate[];
}

export interface DiscoveryImportResponse {
  created: number;
  skipped_duplicate: number;
}

export async function searchDiscovery(query: string, location: string): Promise<DiscoverySearchResponse> {
  return apiFetch<DiscoverySearchResponse>("/discovery/search", {
    method: "POST",
    body: JSON.stringify({ query, location }),
  });
}

export async function importDiscoveryCandidates(
  candidates: DiscoveryCandidate[],
): Promise<DiscoveryImportResponse> {
  return apiFetch<DiscoveryImportResponse>("/discovery/import", {
    method: "POST",
    body: JSON.stringify({ candidates }),
  });
}
